packages/commercetools/api-client/src/index.ts

```diff
@@ -24,26 +24,45 @@ let apolloClient: ApolloClient<any> = null
 let locale = 'en'
 let currency = 'USD'
 let country = 'UK'
+let countries = [{ name: 'US', label: 'United States'}]
+let currencies = [{ name: 'EUR',  label: 'Euro' } ]
+let locales = [{ name: 'en', label: 'English' }]
+let cookies = {
+  currencyCookieName: 'vsf-currency',
+  countryCookieName: 'vsf-country',
+  localeCookieName: 'vsf-locale'
+}
 
 const setup = <TCacheShape>(setupConfig?: SetupConfig<TCacheShape>): ApolloClient<TCacheShape> => {
-  apolloClient = new ApolloClient({
-    link: createCommerceToolsLink(setupConfig.api),
-    cache: new InMemoryCache(),
-    ...setupConfig.customOptions
-  })
-  locale = setupConfig.locale
-  currency = setupConfig.currency
-  country = setupConfig.country
+  if (setupConfig.api) {
+    apolloClient = new ApolloClient({
+      link: createCommerceToolsLink(setupConfig.api),
+      cache: new InMemoryCache(),
+      ...setupConfig.customOptions
+    })
+  }
+
+  locale = setupConfig.locale || locale
```

Comment:
setupConfig can be undefined and this will throw an error `cannot ready property locale of undefined`